src/scripts/main.js

```diff
@@ -1,3 +1,317 @@
+  const tableBody = table.querySelector('tbody');
+
+  tableBody.innerHTML = '';
+
+  sortedData.forEach((rowData) => {
+    const row = document.createElement('tr');
+
+    rowData.forEach((cellData) => {
+      const cell = document.createElement('td');
+
+      cell.textContent = cellData;
+      row.appendChild(cell);
+    });
+    tableBody.appendChild(row);
+  });
+}
+
+function toggleSortDirection() {
+  sortDirection = !sortDirection;
+}
+
+function attachSortingListeners() {
+  table.querySelectorAll('th').forEach((header, index) => {
+    header.addEventListener('click', () => sortTable(index));
+  });
+}
+
+attachSortingListeners();
+
+// 2. When user clicks on a row, it should become selected.
+let activeRow = null;
+const tbody = table.querySelector('tbody');
+
+tbody.addEventListener('click', (e) => {
+  const row = e.target.closest('tr');
+
+  if (!row) {
+    return;
+  }
+
+  if (activeRow) {
+    activeRow.classList.remove('active');
+  }
+
+  row.classList.add('active');
+  activeRow = row;
+});
+
+// 3. Write a script to add a form to the document.
+// Form allows users to add new employees to the spreadsheet.
+const formHtml = `
+  <form class="new-employee-form">
+    <label>
+      Name:
+      <input
+        name="name"
+        type="text"
+        data-qa="name"
+      />
+    </label>
+    <label>
+      Position:
+      <input
+        name="position"
+        type="text"
+        data-qa="position"
+      />
+    </label>
+    <label>
+      Office:
+      <select
+        name="office"
+        data-qa="office"
+      >
+        <option value="Tokyo">Tokyo</option>
+        <option value="Singapore">Singapore</option>
+        <option value="London">London</option>
+        <option value="New York">New York</option>
+        <option value="Edinburgh">Edinburgh</option>
+        <option value="San Francisco">San Francisco</option>
+      </select>
+    </label>
+    <label>
+      Age:
+      <input
+        name="age"
+        type="number"
+        data-qa="age"
+      />
+    </label>
+    <label>
+      Salary:
+      <input
+        name="salary"
+        type="number"
+        data-qa="salary"
+      />
+    </label>
+    <button
+      type="button"
+      id="save-button"
+    >
+      Save to table
+    </button>
+  </form>
+  `;
+
+table.insertAdjacentHTML('afterend', formHtml);
+
+// 4. Show notification if form data is invalid
+function pushNotification(message, type) {
+  const notification = document.createElement('div');
+  const titleSpan = document.createElement('span');
+
+  notification.className = `notification ${type}`;
+  notification.setAttribute('data-qa', 'notification');
+
+  titleSpan.className = 'title';
+  titleSpan.textContent = type.toUpperCase();
+
+  notification.appendChild(titleSpan);
+  notification.innerHTML += ` <span>${message}</span>`;
+  document.body.appendChild(notification);
+
+  setTimeout(() => {
+    notification.style.display = 'none';
+  }, 2000);
+}
```

Comment:
Code quality issue: The `pushNotification` function is responsible for both creating the notification and removing it after a delay. These are two separate concerns and should be handled by separate functions. Also, setting `display` to `none` will not remove the element from the DOM. Consider using `remove` method instead.